martus-client/source/org/martus/client/swingui/dialogs/UiReportFieldChooserDlg.java

```diff
@@ -222,11 +223,25 @@ protected FieldSpecCollection getAllKnownFields()
 	protected void fillSelectorPanel(ChoiceBoxItem selectedFormTemplate)
 	{
 		FieldSpecCollection allFieldsForTemplate = new FieldSpecCollection();
-		allFieldsForTemplate.addAll(selectedFormTemplate.getTopFields());
-		allFieldsForTemplate.addAll(selectedFormTemplate.getBottomFields());
-		
+		FieldSpecCollection topTemplateFields = selectedFormTemplate.getTopFields();
+		FieldSpecCollection bottomTemplateFields = selectedFormTemplate.getBottomFields();
+		allFieldsForTemplate.addAll(topTemplateFields);
+		allFieldsForTemplate.addAll(bottomTemplateFields);
+
+		PoolOfReusableChoicesLists reusableChoicesLists = new PoolOfReusableChoicesLists();
+
+		for (Object name : topTemplateFields.getReusableChoiceNames())
+		{
+			reusableChoicesLists.add(topTemplateFields.getReusableChoices((String) name));
+		}
+
+		for (Object name : bottomTemplateFields.getReusableChoiceNames())
+		{
+			reusableChoicesLists.add(bottomTemplateFields.getReusableChoices((String) name));
+		}
```

Comment:
Extract out a method to that has one for loop and call it twice.   The method would return a PoolOfReusableChoicesLists.  Calling method would call the method twice and create its own PoolOfReusableChoicesLists.  This also requires PoolOfReusableChoicesLists to have an addAll.   Can
we cast the set to Set<String> instead of using Object?